backend/src/index.ts

```diff
@@ -0,0 +1,229 @@
+import express, { Request, Response } from 'express'
+import { prisma } from './database'
+import cors from 'cors'
+import { User } from '@prisma/client'
+
+const Errors = {
+    UsernameAlreadyTaken: 'UserNameAlreadyTaken',
+    EmailAlreadyInUse: 'EmailAlreadyInUse',
+    ValidationError: 'ValidationError',
+    ServerError: 'ServerError',
+    ClientError: 'ClientError',
+    UserNotFound: 'UserNotFound',
+}
+
+const app = express()
+app.use(express.json())
+app.use(cors())
+
+function generateRandomPassword(length: number): string {
+    const charset =
+        'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]{}|;:,.<>?'
+    const passwordArray = []
+
+    for (let i = 0; i < length; i++) {
+        const randomIndex = Math.floor(Math.random() * charset.length)
+        passwordArray.push(charset[randomIndex])
+    }
+
+    return passwordArray.join('')
+}
```

Comment:
I'm leaving my opinion on this because there's nothing wrong with this but we can also use a different approach by removing the generateRandomPassword function altogether and using password String @default(uuid()) in schema.prisma. This way the password will have a default value but it can also be changed